datastore.go

```diff
@@ -388,9 +396,21 @@ func (d *Datastore) Close() error {
 	return d.DB.Close()
 }
 
+// Batch creats a new Batch object. This provides a way to do many writes, when
+// there may be too many to fit into a single transaction.
+//
+// After writing to a Batch, always call Commit whether or not writing to the
```

Comment:
This kind of contract needs to be specified at the interface level to be effective. I think your original solution of adding a `Cancel` method to the interface is the right way to go (we can add it here now, then add it to the interface later.

For now, we could also use `runtime.SetFinalizer` to ensure everything gets cleaned up eventually. The overhead should be relatively minimal.

Reply:
Added a `Cancel` function and finalizer.  Will add Cancel to `datastore.Batch` interface later.